File: mns-core/src/test/java/cz/fb/manaus/core/model/CollectedBetsTest.java
package cz.fb.manaus.core.model;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.Test;

import java.util.Date;

import static org.junit.Assert.assertTrue;

public class CollectedBetsTest {

    @Test
    public void testSerialization() throws Exception {
        var mapper = new ObjectMapper();
        var original = new Bet("111", "222", 333,
                new Price(3d, 2d, Side.BACK), new Date(), 0d);

        var bets = CollectedBets.create();
        bets.getPlace().add(original);
        bets.getUpdate().add(original);
        bets.getCancel().add("100");

        var serialized = mapper.writer().writeValueAsString(bets);
        var tree = mapper.reader().readTree(serialized);
        assertTrue(tree.has("place"));
        assertTrue(tree.has("update"));
        assertTrue(tree.has("cancel"));
    }
}

File: mns-core/src/test/java/cz/fb/manaus/core/model/CollectedBetsTest.kt
package cz.fb.manaus.core.model

import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Test
import java.time.Instant
import kotlin.test.assertTrue

class CollectedBetsTest {

    @Test
    fun serialization() {
        val mapper = ObjectMapper()
        val original = Bet(
                betId = "111", marketId = "222", selectionId = 333,
                requestedPrice = Price(3.0, 2.0, Side.BACK), placedDate = Instant.now()
        )

        val bets = CollectedBets(
                place = listOf(original.asTracked),
                update = listOf(original.asTracked),
                cancel = listOf("100")
        )

        val serialized = mapper.writer().writeValueAsString(bets)
        val tree = mapper.reader().readTree(serialized)
        assertTrue(tree.has("place"))
        assertTrue(tree.has("update"))
        assertTrue(tree.has("cancel"))
    }
}


val Bet.asTracked: TrackedBet
    get() = TrackedBet(
            this,
            betAction.copy(
                    marketId = marketId,
                    selectionId = selectionId,
                    price = requestedPrice
            ),
    )